manifests/broker/service.pp

```diff
@@ -23,6 +26,12 @@
       content => template('kafka/init.erb'),
     }
 
+    if $::osfamily == 'RedHat' and $::operatingsystemmajrelease == '7' {
```

Comment:
This would also be valid on Debian 8 and later Ubuntu. Can you use <code>$::serviceprovider<code> here?